### ISSUE TITLE
Add option to disable automatic dynamic web content for a specific region of a webpage

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#7743]
Steps to reproduce:
1. open this website with NVDA running
http://www.condenast.com/

Expected behavior:
NVDA should stay silent even there is dynamic content on the page.

Actual behavior:
NVDA reads every few seconds the dynamic content. Dynamic content is unchecked under object presentation, as well as tooltips and similar announcements. Animations are turned off in Internet Explorer or Firefox, Accessibility features in Firefox about:config are set true. Jaws does not have this issue.

System configuration:
NVDA version:
2017.3

NVDA Installed or portable:
both

Other information:

Windows version:
Windows 8.1 and Windows 10

Name and version of other software in use when reproducing the issue:
IE11, Firefox 56

### Other questions:

Does the issue still occur after restarting your PC?
yes

Have you tried any other versions of NVDA?
yes, every Version since 2013.2

**Comment by Brian1Gaff** [comment]
I have just tried this in firefox 52 on windows 7 but with the dynamic 
content checked and nothing is being read other than what I go to read.
 What is the difference between 8 and 7 in this respect or is the page only 
sometimes dynamic and I chose not one of those times? grin.

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by Adriani90** [comment]
Yes sometimes is more dynamic. Try to load the page in internet explorer.

 

 

Von: Brian Gaff [mailto:notifications@github.com] 
Gesendet: Sonntag, 12. November 2017 12:03
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Add option to disable automatic dynamic web content for a specific region of a webpage (#7743)

 

I have just tried this in firefox 52 on windows 7 but with the dynamic 
content checked and nothing is being read other than what I go to read.
What is the difference between 8 and 7 in this respect or is the page only 
sometimes dynamic and I chose not one of those times? grin.

bglists@blueyonder.co.uk <mailto:bglists@blueyonder.co.uk> 
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk <mailto:briang1@blueyonder.co.uk> , putting 'Brian Gaff'
in the display name field.
-----

**Comment by Adriani90** [comment]
In firefox seems to depend on a certain plug in which displays the content properly. But I couldn’t found yet which one. It could be any video player or codec.

 

Try to load the page in Internet explorer.

 

Best

Adriani

 

Von: Brian Gaff [mailto:notifications@github.com] 
Gesendet: Sonntag, 12. November 2017 12:03
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Add option to disable automatic dynamic web content for a specific region of a webpage (#7743)

 

I have just tried this in firefox 52 on windows 7 but with the dynamic 
content checked and nothing is being read other than what I go to read.
What is the difference between 8 and 7 in this respect or is the page only 
sometimes dynamic and I chose not one of those times? grin.

bglists@blueyonder.co.uk <mailto:bglists@blueyonder.co.uk> 
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk <mailto:briang1@blueyonder.co.uk> , putting 'Brian Gaff'
in the display name field.
-----

**Comment by Adriani90** [comment]
It seems like NVDA interprets the output of some video codecs different than jaws does. Otherwise I cannot explain why the other screenreader does not have this problem.

 

 

Von: Adriani Botez [mailto:adriani.botez@googlemail.com] 
Gesendet: Sonntag, 12. November 2017 13:45
An: 'nvaccess/nvda' <reply@reply.github.com>; 'nvaccess/nvda' <nvda@noreply.github.com>
Cc: 'Author' <author@noreply.github.com>
Betreff: AW: [nvaccess/nvda] Add option to disable automatic dynamic web content for a specific region of a webpage (#7743)

 

In firefox seems to depend on a certain plug in which displays the content properly. But I couldn’t found yet which one. It could be any video player or codec.

 

Try to load the page in Internet explorer.

 

Best

Adriani

 

Von: Brian Gaff [mailto:notifications@github.com] 
Gesendet: Sonntag, 12. November 2017 12:03
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Add option to disable automatic dynamic web content for a specific region of a webpage (#7743)

 

I have just tried this in firefox 52 on windows 7 but with the dynamic 
content checked and nothing is being read other than what I go to read.
What is the difference between 8 and 7 in this respect or is the page only 
sometimes dynamic and I chose not one of those times? grin.

bglists@blueyonder.co.uk <mailto:bglists@blueyonder.co.uk> 
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk <mailto:briang1@blueyonder.co.uk> , putting 'Brian Gaff'
in the display name field.
-----

**Comment by PratikP1** [comment]
I have not been able to replicate this behavior in Firefox 56 or Nightly. I was able to do so in IE. I tested with both dynamic content enabled and disabled. It appears to make no difference. I rather suspect that this issue is related to the particular web site rather than NVDA’s behavior.

**Comment by Adriani90** [comment]
I think rather that it is the way NVDA interacts with video codecs. IE seems to display the content properly due to its codecs. But other screen readers do not have this problem.

 

Von: PratikP1 [mailto:notifications@github.com] 
Gesendet: Sonntag, 12. November 2017 13:58
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Add option to disable automatic dynamic web content for a specific region of a webpage (#7743)

 

I have not been able to replicate this behavior in Firefox 56 or Nightly. I was able to do so in IE. I tested with both dynamic content enabled and disabled. It appears to make no difference. I rather suspect that this issue is related to the particular web site rather than NVDA’s behavior.

**Comment by Brian1Gaff** [comment]
Well, what can I say, I have no extra things loaded and I do use the klite 
codec pack but that is really only relevant on embedded or rich content I'd 
have thought. The rest of the time its using the generic loaded with firefox 
extensions and add ons.All I have are the flash and normal ones and in the 
add onse  navigational sounds and the ublock Origin advert remover.




bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by derekriemer** [comment]
Are you asking for a feature to make live regions on websites (Dynamic content which speaks as info changes) be controlled by the dynamic content changes feature?

**Comment by Brian1Gaff** [comment]
In the original thread which was on the user list the person complained that 
a particular page kept speaking the active area no matter what switches were 
on or off. as it was a page internal to his company, nobody could test it, 
so somebody then supplied a page where it happened but for me at least it 
was fine, and yet others were saying it was r3epeating some info every few 
seconds.
 I think its going to need more investigation, perhaps looking at the 
dynamic content thread on the .io list for some inspiration.
 To me it sounds  like one of two things. Either his company has made a page 
that refreshes  in a non standard way, or some peoples machines and browsers 
actually are set up differently.
 Who knows.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by Adriani90** [comment]
That with the company page was me indeed. We implemented a sharepoint solution at work and the behavior is the same as on the website above. It is the same one which I have posted in the treat. at work I have the problem in Firefox eather, but not at home. At home the problem still persists in Internet explorer.

It occurs only with NVDA.

On the sharepont site there is exactly the same kind of embeded element like on the website above.

Derek, if that feature would solve my problem, than yes. My problem is that NVDA speaks those information avery time even when I change the window. In the example of the sharepoint site, there is an embeded element which shows the updated time zone of every country we are active in. This update take place every second so you can imagine how NVDA speaks in my ears.

**Comment by Adriani90** [comment]
Sorry for the spell errors guys, I am little bit in hurry.

Thanks for your help.

**Comment by Brian1Gaff** [comment]
The weird thing is that in firefox on that public page the embedded element 
seems to be totally empty to me, and one has to get out of it to continue to 
use the page. I'll try the IE  one, but this seems to me to be that we are 
missing something somewhere here, but cannot pin it down.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by Adriani90** [comment]
So, I've finally found the trigger for this. Java seems to be the problem again. If your Java works properly in Firefox, then you will have this problem in Firefox as well. I have disabled Java in about:config and there was no problem anymore. I have enabled it again and NVDA begins speaking all the time. It seems to be a problem in NVDA regarding Java script in browsers. I have searched in about:config for "Java" and with Java script on I have tried to change setting for setting to discover which one triggers the problem. But there are many settings which I don't know how to change in order to see differences. I couldn't find any setting in Java which would solve this problem by changing it. It has to be something in the NVDA Code I think. But I would very appreciate if someone can have a look and try to find a solution.

**Comment by Brian1Gaff** [comment]
That makes sense as this machine has no Java installed on it. Its more 
trouble than it is worth. I remember on XP I was forever being told that 
Java was insecure and the  add on was disabled, so what is the point I 
thought.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by Brian1Gaff** [comment]
Are you confusing javascript with Java though. its not entirely clear as 
I've most certainly not changed javascript in firefox and its on by default.

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by Adriani90** [comment]
I'm still having this problem. It is related to the java plugin. Other screen readers do not have this problem. Do you have any suggestions how to solve it?

**Comment by Adriani90** [comment]
#1318 is related.

**Comment by Adriani90** [comment]
#9634 partly improves the experience with NVDA and live regions in Firefox. Now NVDA reports them only when your focus is in the coresponding tab. This is indeed very appreciated.

Now the problem still exists in IE and in corporate environments many SAP or sharepoint solutions work only in IE. Live regions are reported in IE although I don't focus the window at all. Even when I am on desktop the live region of the website is reported. So it would still be very appreciated if someone could add an option to document formating to disable reporting of live regions entirely. I don't hink that a solution for IE will be provided but such an option to disable or enable live region reporting could help in many other circumstances and also in my case.
@derekriemer will your PR #9079 also add such an option?
CC: @feerrenrut

**Comment by JulienCochuyt** [comment]
@derekriemer, I'm not sure to understand how exactly does PR #9079 fixes this very issue.
It for sure made things better, but if the main goal is correctly stated in the title "Add option to disable automatic dynamic web content for a specific region of a webpage", then this goal is not fulfilled as `nvdaControllerInternal_reportLiveRegion` receives no information about which part of the document triggered the announce.
Did I miss some new mechanism to specify upfront which source we do not want live announce from?

@feerrenrut, re-opening until clarified.

**Comment by JulienCochuyt** [comment]
Likely related: #12193

**Comment by Adriani90** [comment]
This is not reproducible anymore as it stands, I am closing as works for me.
